### ISSUE TITLE
shfmt: Allow short multiple statements in subshells

### ISSUE THREAD

**Issue by brandonkal** [mvdan/sh#591]
This
```sh
ver=$(set -e; missing v)
```
becomes this
```sh
ver=$(
	set -e
	missing v
)
```
I'd argue that the first is preferable formatting. In the same way:
```sh
ver=$(
	missing v
)
```
should collapse to
```sh
ver=$(missing v)
```

**Comment by mvdan** [comment]
Regarding the first case, see #261, which has some history.

Regarding the second, I'm not sure. I think it falls under "collapsing multiple lines into a single short line", which is very subjective. What if, instead, you had:

```
some_very_long_variable_name=$(
	some program with a very very very long list of arguments
)
```

I would argue this is easier to read and maintain as multiple lines, and more so if the length increases further.

**Comment by mvdan** [comment]
There wasn't a response here, and we closed #261 in favor of #564, which is about an actual objective benefit - security. So I'm closing this as a duplicate of #261.
